forest/src/daemon.rs

```diff
@@ -335,20 +338,12 @@ pub(super) async fn start(config: Config, detached: bool) {
 
     // Terminate if no snapshot is provided or DB isn't recent enough
     match chain_store.heaviest_tipset().await {
-        None => {
-            cli_error_and_die(
-                "Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file]",
-                1,
-            );
-        }
+        None => prompt_and_fetch_snapshot(&mut config).await,
```

Comment:
I like this idea, but then we run into the issue of not having RPC up while fetching the snapshot. Just like we had before when downloading headers. I don't think this is an issue that needs to be addressed but it will only apply when a snapshot is being downloaded. Looking at the code, it is possible to just stick it right before the p2p service starts up. I'll make the change but open to discussion about this

Reply:
Either this or we would need some locking on `stdout` or pausing other tasks. 

A simple approach I can think of is having all interactive prompts before all other tasks - I don't expect forest to have RPC server running while the application is in babysitter mode. But we need to ensure this is the case across the entire codebase.

Thoughts? @lemmih @elmattic @tyshko5 @jdjaustin @connormullett ?

Reply:
@LesnyRumcajs I'm looking at it now.

Reply:
Exactly Hubert, all interactive prompts before the daemon starts (in detached mode or not).

Reply:
@connormullett We don't need forest daemon RPC service for running the `snapshot_fetch` util (he internaly makes use of `reqwest` or `s3` crates).